NHSN/src/test/java/com/nhsn/pages/ReviewPage.java

```diff
@@ -54,6 +59,36 @@ public void clickOnGenerateButton()
         System.out.println("Clicked on Generate Button");
     }
 
+    public void enterSubmitDateOnReviewPage(String date) throws InterruptedException {
+
+        WebElement submitDateTextbox = driver.findElement(By.xpath("//input[@name='dp']"));
+        char dateArray[] = date.toCharArray();
+        for(int i=0;i< dateArray.length;i++)
+        {
+            submitDateTextbox.sendKeys(Character.toString(dateArray[i]));
+            Thread.sleep(1000);
+        }
+        System.out.println("Entered the date as -"+ date);
+    }
+
+    public void verifySubmittedReportOnReviewPage( String note)  {
+
+        WebElement reportNote = driver.findElement(By.xpath("//table//span[text()='"+note+"']"));
+        Assert.assertTrue(reportNote.isDisplayed(), "Recently Submitted Report is not displayed on the review Page");
+        System.out.println(note + " is displayed as expected i.e. Recently Submitted Report is displayed on the review Page");
+    }
+
+    public String getTooltipOfSubmittedDate(String note) throws InterruptedException {
+
+        WebElement submittedDateElement = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//table//span[text()='"+note+"']/ancestor::tr/td[4]/span")));
+        Actions action = new Actions(driver);
+        action.moveToElement(submittedDateElement).build().perform();
+        WebElement tooltipElement = driver.findElement(By.xpath("//ngb-tooltip-window[@role='tooltip']"));
```

Comment:
Looks like this will find any tooltip in the whole page. Consider creating a more specific xpath/css, as we may in the future have more tooltips on the page.

Reply:
Before we find the tooltip, we are mouse hovering on the Webelement  then we are finding the Tooltip. If we dont mouse hover on the element, we are showing the Tooltip properties . So we should be good here.